### PR TITLE
ENT-250 Introduce catalog api endpoints in enterprise api.yaml

### PR DESCRIPTION
**Pull request by asadiqbal08**
**Description:**  Introduce catalogs API endpoints in enterprise api.yaml file for amazon swagger 

@mattdrayer @saleem-latif This is a WIP PR to get feedback. It is fully dependent on [PR#62](https://github.com/edx/edx-enterprise/pull/62/)

**JIRA:** [ENT-250](https://openedx.atlassian.net/browse/ENT-250)

**Merge checklist:**

- [x] All reviewers approved
- [x] CI build is green
- [x] Version bumped
- [x] Changelog record added
- [x] Commits are (reasonably) squashed
- [x] PR author is listed in AUTHORS

### CODE REVIEW

**Comment by asadiqbal08** [comment]
We need to create the corresponding swagger mapping(s) in [api-manager](https://github.com/edx/api-manager/blob/master/swagger/api.yaml)

**Comment by mattdrayer** [comment]
This is a good start -- I am assuming we will also need to include an `api-compact.yaml` as well.  What other work do you feel is necessary for this PR before we can add it to the project?

**Comment by asadiqbal08** [comment]
@mattdrayer I am curious to know the purpose of api-compact.yaml file. I can see [API-Manager](https://github.com/edx/api-manager/blob/master/swagger/api.yaml#L33) make a reference to api.yaml. 
I think, We are ok with that change sets at that point.

**Comment by mattdrayer** [comment]
👍

File: api-compact.yaml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+            uri: "https://${stageVariables.enterprise_host}/v1/catalogs/{id}/"
+
+    # /v1/catalogs/{id}/courses
+    catalogCourses:
+        get:
+          produces: *produces
+          parameters:
+            - *auth_header
+            - *id_parameter
+            - *limit_qs_parameter
```
Comment (saleem-latif) [inline api-compact.yaml]: for catalog corses, we are using paginated response as opposed to `limit-offset` response. So, we need to update also allow `page` parameter whose value will be int. I have not tested it properly although I am sure  `limit-offset` style will also work. Any way we still need to add `page` query string parameter.

File: api.yaml [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+            name: Authorization
+            required: true
+            type: string
+          -
+            in: path
+            name: id
+            required: true
+            type: number
+          -
+            in: querystring
```
Comment (saleem-latif) [inline api.yaml]: same as above, we can leave `limit-offset` parameters as is too, but we need `page` qs parameter here.

**Comment by asadiqbal08** [comment]
@mattdrayer Should I merge it ?

**Comment by mattdrayer** [comment]
Yep, two thumbs, ready to go!

**Pull request merged**